### ISSUE TITLE
'Not Found' error when using file select pop-up

### ISSUE THREAD

**Issue by JustinTBrown** [colymba/GridFieldBulkEditingTools#125]
Hi, when attempting to use the bulk upload function together with GridFieldGalleryTheme I'm getting the following error when using the pagination controls to view available images:
![screen shot 2016-04-03 at 10 07 13 pm](https://cloud.githubusercontent.com/assets/1620519/14237993/8147183c-f9e8-11e5-90fb-18a52adb7b66.png)

Which after clearing is followed by:
![screen shot 2016-04-03 at 9 34 03 pm](https://cloud.githubusercontent.com/assets/1620519/14237934/c045699a-f9e7-11e5-9d53-929e4fccdaec.png)

My GalleryPage is setup as follows:
```
class GalleryPage extends Page {
  private static $has_many = array(
    'GalleryImages' => 'GalleryImage',
  );  
  public function getCMSFields() {
    $fields = parent::getCMSFields();
    $fields->removeByName('Content');
    $fields->addFieldsToTab('Root.Main',array(
      $gridfield = GridField::create('GalleryImages', 'Gallery', $this->GalleryImages())
    ));
    $config = $gridfield->getConfig();
    $config->addComponent(new GridFieldGalleryTheme('Image'));
    $config->addComponent(new GridFieldBulkUpload());
    $gridfield->setConfig($config);
    return $fields;
  }
}
```

And my GalleryImage is:
```
class GalleryImage extends DataObject {
  public static $has_one = array(
    'GalleryPage' => 'GalleryPage',
    'Image' => 'Image',
  );
}
```

I'm able to add images from the first page of results, but can not access the the next 8 pages of results. I'm also able to add single images if I use the GridFieldConfig_RecordEditor (which I've removed from the above example for brevity). I also get the same result with GridFieldGalleryTheme commented out.

I'm using SS3.2, and GridFieldBulkEditing 2.1.5 (and initially had GridFieldBulkEditing "dev-master" installed with the same issue). Am I missing anything obvious? Been bumping my head against a wall with this one for a while.

**Comment by colymba** [comment]
@JustinTBrown not your fault and not because using both modules together :smile:

The attach from file is broken.... This is a known issue that keeps creeping back... same as here https://github.com/colymba/GridFieldBulkEditingTools/issues/124 the `UploadField` urls need to be handled better so they redirect through the `GridField`...

**Comment by JustinTBrown** [comment]
@colymba beautiful. Thanks for the heads up! I wondered if it was similar #124. Any sense of when 2.2 is coming? Weeks? Months? Just trying to decide if I should hold out for a fix or figure out another solution.

**Comment by colymba** [comment]
@JustinTBrown I've marked 2.2 for July... that's a long way from now but that's for a bunch of issues to be resolved... and that https://github.com/colymba/GridFieldBulkEditingTools/issues/124 is a big one that needs fixing asap so it's first on my list....

Unless something unexpected I don't see why this couldn't be fixed in the coming couple of weeks when I have some time and make it in a 2.16 release.....

**Comment by JustinTBrown** [comment]
Ok, thanks! Really appreciate the update. :)

On Tue, Apr 5, 2016 at 4:27 AM, Thierry François <notifications@github.com> wrote:
@JustinTBrown [https://github.com/JustinTBrown] I've marked 2.2 for July... that's a long way from now but that's for a bunch of issues to be resolved... and that #124 [https://github.com/colymba/GridFieldBulkEditingTools/issues/124] is a big one that needs fixing asap so it's first on my list....

Unless something unexpected I don't see why this couldn't be fixed in the coming couple of weeks when I have some time and make it in a 2.16 release.....

**Comment by colymba** [comment]
This should be fixed in master now with 29482dbd17e42c3de364aa3d46d20ba65175c9d3
Try it out and let me know if not.
